src/backend/distributed/metadata/node_metadata.c

```diff
@@ -483,11 +485,16 @@ citus_disable_node(PG_FUNCTION_ARGS)
 		ereport(ERROR, (errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
 						errmsg("disabling the first worker node in the "
 							   "metadata is not allowed"),
-						errhint("You can force disabling node, but this operation "
-								"might cause replicated shards to diverge: SELECT "
-								"citus_disable_node('%s', %d, force:=true);",
-								workerNode->workerName,
-								nodePort)));
+						errhint("You can force disabling node, SELECT "
+								"citus_disable_node('%s', %d, force:=true, "
+								"synchronous:=true); "
+								"Passing synchronous:=false might cause replicated shards "
```

Comment:
This message is a bit awkward, should we disallow synchronous:=false maybe?

Reply:
Well, agree that the message is not nice.

Also, I think I allowed `force:=true AND synchronous:=false`, because it allows disabling the first worker node in case there are multiple node failures and you still want to proceed at your own risk. 

Do you suggest to prevent  `force:=true AND synchronous:=false`? If not, I can work on nicer wording for the message

Reply:
> Also, I think I allowed force:=true AND synchronous:=false, because it allows disabling the first worker node in case there are multiple node failures and you still want to proceed at your own risk.

wouldn't you be able to disable other nodes first?

thinking about this more: what happens if we have asynchronously disabled some nodes, and then synchronously disable a node? to which nodes do we propagate?

Reply:
> thinking about this more: what happens if we have asynchronously disabled some nodes, and then synchronously disable a node? to which nodes do we propagate?

Coordinator marks node disabled immediately via `SetWorkerColumnLocalOnly()`, and we would only send the active ones that the coordinator knows at that time. So, async disabled nodes would not be synced again.

> wouldn't you be able to disable other nodes first?

Yes, you can actually. So then `force:=true AND synchronous:=false` becomes not a very useful combination. I can prevent as you suggested earlier.

Reply:
With this new perspective, `force` mode becomes irrelevant. So, I'm leaning towards dropping `force` but only keep `synchronous` option. In the end, it (a) prevents data divergence opportunity when first node is disabled (b) provides a reasonable option for disabling a node when there are no other node failures (e.g., maintenance of a single node)  (c) a lot more straightforward to reason about the parameter -- compared to `force`

I think the only slight disadvantage is that, when there are multiple node failures, and user wants to disable worker-1 and worker-2, she should first disable worker-2 in async mode, and then worker-1 sync mode. This is not very trivial to come up with. Still, I can document it in the PR/issue description, which at least gives pointers for the users. (and even consider opening a PR on docs repo) 

Let me know if you agree, if so I can adjust the PR description / commit messages as well.